src/simulator_core/entities/heat_network.py

```diff
@@ -36,11 +37,22 @@ def __init__(self, conversion_factory: Callable[[pandapipesNet], Tuple[List[Asse
     def run_time_step(self, time: float, controller_input: dict) -> None:
         """Method to simulate a time step.
 
+        It first sets the controller input to the assets and then simulates the time step.
+
         :param float time: Timestep for which to simulate the model
         :param dict controller_input: Dict specifying the heat demand for the different assets.
         :return: None
         """
-        pass
+        for py_asset in self.assets:
+            if py_asset.asset_id in controller_input:
+                py_asset.set_setpoints(controller_input[py_asset.asset_id])
+        try:
```

Comment:
Why the try pass loop? What does it try and when does it fail?

Reply:
Next line is # pipeflow(self.panda_pipes_net, "all") This lien can be used when all
If we simulate now it will throw an error, when we can simualte we can enable this line and then the try makes sense

Reply:
TODO added